src/main/java/com/careerfair/q/service/database/implementation/StudentFirebaseImpl.java

```diff
@@ -110,4 +115,56 @@ public Student getStudentWithEmail(String email) throws ExecutionException,
 
         throw new FirebaseException("No student with email=" + email + " exists");
     }
-}
+
+    @Override
+    public Student updateStudent(String studentId, Map<String, Object> updatedValues) throws ExecutionException,
+            InterruptedException {
+
+        Firestore firestore = FirestoreClient.getFirestore();
+        DocumentSnapshot document = firestore.collection(STUDENT_COLLECTION)
+                .document(studentId).get().get();
+        if (!document.exists()) {
+            throw new FirebaseException("No student with student id=" + studentId + " exists");
+        }
+        Student student = document.toObject(Student.class);
+        assert student != null;
+
+        updatedValues.forEach(
+                (field, v) -> {
+                    switch (field) {
+
+                        // TODO: What if university_id/student_id is attempted to be modified?
+
+                        // Since these attributes are mapped to camel case, have to separate them out
+                        // Not the best way since it's 'hardcoded'
+                        case "last_name": student.setLastName((String) v); break;
+                        case "first_name": student.setFirstName((String) v); break;
+                        case "grad_date":
+                            LinkedHashMap tmp = (LinkedHashMap) v;
+                            Integer seconds = (Integer) tmp.get("seconds");
+                            Integer nanos = (Integer) tmp.get("nanos");
+                            Timestamp grad = Timestamp.ofTimeSecondsAndNanos(seconds.longValue(), nanos);
+                            student.setGraduationDate(grad);
+                            updatedValues.put("grad_date", grad);
+                            break;
+                        default:
+                            Field studentField = ReflectionUtils.findField(Student.class, field);
+                            try {
+                                studentField.setAccessible(true);
+                            } catch (NullPointerException ex) {
+                                throw new IllegalArgumentException("Invalid field name provided: " + field);
+                            }
+                            ReflectionUtils.setField(studentField, student, v);
```

Comment:
I don't think this is the right place to use reflection. Using reflection should be avoided